pkg/console/controllers/service/controller.go

```diff
@@ -0,0 +1,170 @@
+package service
+
+import (
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformersv1 "k8s.io/client-go/informers/core/v1"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	operatorclientv1 "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	"github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/console/status"
+	"github.com/openshift/console-operator/pkg/console/subresource/service"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/management"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+)
+
+const (
+	// key is basically irrelevant
+	controllerWorkQueueKey = "service-sync-work-queue-key"
+	controllerName         = "ConsoleServiceSyncController"
+)
+
+// ctrl just needs the clients so it can make requests
+// the informers will automatically notify it of changes
+// and kick the sync loop
+type ServiceSyncController struct {
+	operatorConfigClient operatorclientv1.ConsoleInterface
+	// live clients, we dont need listers w/caches
+	serviceClient coreclientv1.ServicesGetter
+	// names
+	targetNamespace string
+	serviceName     string
+	// events
+	cachesToSync []cache.InformerSynced
+	queue        workqueue.RateLimitingInterface
+	recorder     events.Recorder
+}
+
+// factory func needs clients and informers
+// informers to start them up, clients to pass
+func NewServiceSyncController(
+	operatorConfigClient operatorclientv1.ConsoleInterface,
+	corev1Client coreclientv1.CoreV1Interface,
+	serviceInformer coreinformersv1.ServiceInformer,
+	// names
+	targetNamespace string,
+	serviceName string,
+	// events
+	recorder events.Recorder,
+) *ServiceSyncController {
+
+	corev1Client.Services(targetNamespace)
+
+	ctrl := &ServiceSyncController{
+		operatorConfigClient: operatorConfigClient,
+		serviceClient:        corev1Client,
+		// names
+		targetNamespace: targetNamespace,
+		serviceName:     serviceName,
+		// events
+		recorder:     recorder,
+		cachesToSync: nil,
+		queue:        workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), controllerName),
+	}
+
+	serviceInformer.Informer().AddEventHandler(ctrl.newEventHandler())
```

Comment:
shouldn't we add `operatorConfigInformer` as well, to trigger a sync loop after config gets updated ?

Reply:
We do not want informers for anything but `service` for this one.  
That was a mistake on my part to suggest adding them to your `Downloads` controller, we can follow & remove.  
The theory is that it can hide bugs by having unrelated resource kick your sync when it shouldn't sync.